service/history/transferQueueProcessor.go

```diff
@@ -289,6 +296,11 @@ func (t *transferQueueProcessorImpl) completeTransferLoop() {
 				err := t.completeTransfer()
 				if err != nil {
 					t.logger.Info("Failed to complete transfer task", tag.Error(err))
+					if err == ErrShardClosed {
+						// shard closed, trigger shutdown and bail out
+						t.Stop()
```

Comment:
Is this now unrecoverable?

Reply:
Even before it was unrecovered.  Once the shard is stolen underneath queue processor the only thing it can do is shutdown.